src/main/java/org/openbase/bco/bcozy/controller/GroupController.java

```diff
@@ -0,0 +1,123 @@
+package org.openbase.bco.bcozy.controller;
+
+import de.jensd.fx.glyphs.fontawesome.FontAwesomeIcon;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.collections.ObservableList;
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.control.*;
+import javafx.scene.layout.Pane;
+import javafx.util.Callback;
+import org.openbase.bco.bcozy.model.LanguageSelection;
+import org.openbase.bco.bcozy.util.Groups;
+import org.openbase.bco.bcozy.view.Constants;
+import org.openbase.bco.bcozy.view.ObserverButton;
+import org.openbase.bco.bcozy.view.SVGIcon;
+import rst.domotic.unit.UnitConfigType;
+
+/**
+ * @author vdasilva
+ */
+public class GroupController {
+
+    @FXML
+    private ObserverButton saveButton;
+    @FXML
+    private TextField label;
+    @FXML
+    private TableColumn<UnitConfigType.UnitConfig, String> removeColumn;
+
+    @FXML
+    private TableColumn<UnitConfigType.UnitConfig, String> nameColumn;
+
+    @FXML
+    private TableView<UnitConfigType.UnitConfig> groupsTable;
+
+    @FXML
+    private Pane root;
+
+    private ObservableList<UnitConfigType.UnitConfig> groups = Groups.getGroups();
+
+    @FXML
+    public void initialize() {
+
+        saveButton.setApplyOnNewText(String::toUpperCase);
+
+        nameColumn.setCellValueFactory(param -> new SimpleStringProperty(param.getValue().getLabel()));
+        LanguageSelection.addObserverFor("groupLabel", (locale, text) -> nameColumn.setText(text));
+
+
+        removeColumn.setCellFactory(cellFactory());
+
+
+        groupsTable.widthProperty().addListener((observable, oldValue, newValue) ->
+                nameColumn.setPrefWidth(newValue.doubleValue() - removeColumn.getWidth() - 2)
+        );
+
+        groupsTable.setEditable(false);
+        nameColumn.setResizable(false);
+        removeColumn.setResizable(false);
+        groupsTable.setItems(groups);
+
+        LanguageSelection.addObserverFor("groupLabel", (locale, text) -> label.setPromptText(text));
+    }
+
+    public Pane getRoot() {
+        return root;
+    }
+
+    private Callback<TableColumn<UnitConfigType.UnitConfig, String>, TableCell<UnitConfigType.UnitConfig, String>> cellFactory() {
+        return new Callback<TableColumn<UnitConfigType.UnitConfig, String>, TableCell<UnitConfigType.UnitConfig, String>>() {
+            @Override
+            public TableCell<UnitConfigType.UnitConfig, String> call(final TableColumn<UnitConfigType.UnitConfig, String> param) {
+                final TableCell<UnitConfigType.UnitConfig, String> cell = new TableCell<UnitConfigType.UnitConfig, String>() {
+
+                    final Button btn = new Button();
+
+                    {
+                        btn.setGraphic(new SVGIcon(FontAwesomeIcon.TIMES, Constants.EXTRA_SMALL_ICON, true));
```

Comment:
i have never seen such a syntax in java. Whats that for?

Reply:
It's an instance initializer 'cause you can't declare constructors in anonymous classes